ui/src/components/FacetsGrid.js

```diff
@@ -1,26 +1,94 @@
 import './FacetsGrid.css';
-
-import React from 'react';
 import FacetCard from "./FacetCard";
+import ApiClient from "../api/src/ApiClient";
+import FacetsApi from "../api/src/api/FacetsApi";
+
+import React, { Component } from 'react';
 import {GridList, GridTile} from "material-ui/GridList";
 
-function FacetsGrid(props) {
-    const facetsResponse = props.facetsResponse;
-    const facetsList = facetsResponse.facets.map((facet) =>
-        <GridTile key={facet.name}>
-            <FacetCard facet={facet} count={facetsResponse.count} className='facetCard'/>
-        </GridTile>
-    );
-    return (
-        <GridList
-            className="gridList"
-            cols={3}
-            cellHeight="auto"
-            padding={1}
-        >
-            {facetsList}
-        </GridList>
-    );
+class FacetsGrid extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            facetsResponse: props.facetsResponse
+        };
+        this.searchFilters = new Map();
+        let apiClient = new ApiClient();
+        apiClient.basePath = '/api';
+        this.api = new FacetsApi(apiClient);
+        this.callback = function(error, data) {
+            if (error) {
+                console.error(error);
+                // TODO(alanhwang): Redirect to an error page
+            } else {
+                this.setState({facetsResponse: data});
```

Comment:
Looking at React examples, usually you extract things from response, eg from https://reactjs.org/docs/faq-ajax.html:
```
componentDidMount() {
    fetch("https://api.example.com/items")
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            isLoaded: true,
            items: result.items
          });
        },
```
Conceptually, state is a list of facets. We don't care that we got the facets from an XHR.

Reply:
I mean, instead of
```
          this.setState({facetsResponse: data});
```
How about
```
          this.setState({
            totalCount: data.count,
            facets: data.facets
          });
```

Reply:
Makes sense, done.